### PR TITLE
Add long polling support

### LINKED ISSUE #648: Add long polling support
Recently SAPI added support for long polling calls on problem status endpoints.

Long polling behavior is activated by appending `timeout=<sec>` query parameter.

Default value for timeout is 1 second, and maximum allowed value is 30 seconds.

In addition to short polling with exponential back-off, we want to add long-polling as polling schedule strategy, and eventually make it the default.

### PR DESCRIPTION
**Pull request by randomir**
Close #648.

### CODE REVIEW

File: dwave/cloud/client/base.py [hunk 1]
```diff
@@ -1449,9 +1453,16 @@ def _do_cancel_problems(self):
...
 
-    def _poll(self, future):
+    def _poll(self, future: Future) -> None:
         """Enqueue a problem to poll the server for status."""
 
+        # split for simplicity
+        if self.config.polling_schedule.strategy == PollingStrategy.BACKOFF:
+            return self._poll_using_backoff(future)
+        else:
+            return self._poll_using_long_polling(future)
```
Comment (arcondello) [inline dwave/cloud/client/base.py:1463]:
```python
        if self.config.polling_schedule.strategy == PollingStrategy.BACKOFF:
            return self._poll_using_backoff(future)
        elif self.config.polling_schedule.strategy == PollingStrategy.LONG_POLLING:
            return self._poll_using_long_polling(future)
        else:
            raise RuntimeError("unexpected polling strategy")
```
might make future development a tiny bit easier

File: dwave/cloud/config/models.py [hunk 2]
```diff
@@ -101,6 +111,19 @@ class PollingSchedule(BaseModel, GetterMixin):
...
 
 
+class LongPollingSchedule(BaseModel):
+    """Problem status long polling params."""
+
+    #: Long polling strategy
+    strategy: Literal[PollingStrategy.LONG_POLLING] = PollingStrategy.LONG_POLLING
+
+    #: Maximum duration long polling connection is kept open, in seconds.
+    wait_time: Optional[int] = 30
```
Comment (arcondello) [inline dwave/cloud/config/models.py:121]: Feels slightly strange to have this as an `int` and `pause` as a `float`.

**Review by arcondello — APPROVED** [review]
Gave it a quick look. Will take another deeper look on a second pass, but if this is urgent is OK to merge sooner.

File: dwave/cloud/config/models.py (hunk 2 above)
Comment (randomir) [reply dwave/cloud/config/models.py:121]: I agree, but SAPI requires `int` for `timeout` (not sure why). And `pause` is a local sleep (we'll probably remove it eventually, as I don't see many benefits/use cases).

Comment (arcondello) [reply dwave/cloud/config/models.py:121]: Might be worth a comment? :shrug:

File: dwave/cloud/client/base.py (hunk 1 above)
Comment (randomir) [reply dwave/cloud/client/base.py:1463]: Actually had it like that initially, but simplified later. But sure, can add it back, better to explicitly fail.

**Review by arcondello — APPROVED** [review]
Looks good on a second look as well!

**Pull request merged**